dll/win32/shell32/CDefaultContextMenu.cpp

```diff
@@ -1259,6 +1277,40 @@ CDefaultContextMenu::InvokeRegVerb(
 
     PStaticShellEntry pEntry = &m_StaticEntries.GetAt(it);
 
+    CRegKey VerbKey;
+    WCHAR VerbKeyPath[sizeof("shell\\") + MAX_VERB];
+    hr = StringCbPrintfW(VerbKeyPath, sizeof(VerbKeyPath), L"shell\\%s", pEntry->Verb.GetString());
+    if (SUCCEEDED(hr) && m_pDataObj &&
+        VerbKey.Open(pEntry->hkClass, VerbKeyPath, KEY_READ) == ERROR_SUCCESS)
+    {
+        CLSID clsid;
+
+        DWORD KeyState = 0;
+        if (lpcmi->fMask & CMIC_MASK_SHIFT_DOWN)
+            KeyState |= MK_SHIFT;
+        if (lpcmi->fMask & CMIC_MASK_CONTROL_DOWN)
+            KeyState |= MK_CONTROL;
+
+        POINTL *pPtl = NULL;
+        C_ASSERT(sizeof(POINT) == sizeof(POINTL));
+        if (lpcmi->fMask & CMIC_MASK_PTINVOKE)
+            pPtl = (POINTL*)&lpcmi->ptInvoke;
+
+        // TODO: IExecuteCommand
+
+        CComPtr<IDropTarget> pDT;
+        hr = SHELL_GetRegCLSID(VerbKey, L"DropTarget", L"CLSID", clsid);
+        if (SUCCEEDED(hr))
+            hr = CoCreateInstance(clsid, NULL, CLSCTX_ALL, IID_PPV_ARG(IDropTarget, &pDT));
+        if (SUCCEEDED(hr))
+        {
+            IUnknown_SetSite(pDT, static_cast<IContextMenu*>(this));
+            hr = SHSimulateDrop(pDT, m_pDataObj, KeyState, pPtl, NULL);
+            IUnknown_SetSite(pDT, NULL);
```

Comment:
Last-time review question: Isn't it possible to use `pDT->SetSite(...)` here and above?

Reply:
IDropTarget does not have that method, you have to QueryInterface for one of the site interfaces, that is why that helper function exists.

Reply:
makes sense :+1: